File: backend/apps/usuarios/models/notificacion.py
from django.db import models

from ..data.tipos_notificacion import TipoNotificacion
from .reserva import Reserva
from .usuario import Usuario


class Notificacion(models.Model):
    usuario = models.ForeignKey(
        Usuario,
        on_delete=models.CASCADE,
        related_name="notificaciones"
    )

    reserva = models.ForeignKey(
        Reserva,
        on_delete=models.CASCADE,
        related_name="notificaciones",
        null=True,
        blank=True
    )

    tipo = models.CharField(
        max_length=30,
        choices=TipoNotificacion.choices
    )

    mensaje = models.TextField()

    leida = models.BooleanField(default=False)

    creada_en = models.DateTimeField(
        auto_now_add=True
    )

    def __str__(self):
        return f"{self.usuario.username} - {self.tipo}"

File: backend/src/models/Notificacion.ts
import { Schema, model, Document, Types } from "mongoose";
import { TIPOS_NOTIFICACION, TipoNotificacion } from "../data/tiposNotificacion";

export interface INotificacion extends Document {
  usuario: Types.ObjectId;
  reserva?: Types.ObjectId;
  tipo: TipoNotificacion;
  mensaje: string;
  leida: boolean;
  creadaEn: Date;
}

const notificacionSchema = new Schema<INotificacion>(
  {
    usuario: { type: Schema.Types.ObjectId, ref: "Usuario", required: true },
    reserva: { type: Schema.Types.ObjectId, ref: "Reserva" },
    tipo: { type: String, enum: Object.values(TIPOS_NOTIFICACION), required: true },
    mensaje: { type: String, required: true },
    leida: { type: Boolean, default: false },
  },
  { timestamps: { createdAt: "creadaEn", updatedAt: false } }
);

export const Notificacion = model<INotificacion>("Notificacion", notificacionSchema);
